crates/subspace-farmer/src/bin/subspace-farmer/commands/farm.rs

```diff
@@ -17,7 +18,23 @@ use subspace_networking::Config;
 use subspace_rpc_primitives::FarmerProtocolInfo;
 use tracing::{info, trace, warn};
 
-use crate::{ArchivingFrom, DiskFarm, FarmingArgs};
+struct CallOnDrop(Option<Box<dyn FnOnce() + Send + 'static>>);
```

Comment:
Btw, we can just use generic here and just unbox it

Reply:
Indeed, applied in latest commit